astropy/modeling/tests/test_models.py

```diff
@@ -252,7 +252,10 @@ def test_bounding_box2D(self, model_class, test_parameters):
         outside_bbox = model(x2, y2, with_bounding_box=True)
         outside_bbox = outside_bbox[~np.isnan(outside_bbox)]
 
-        assert np.all(inside_bbox == outside_bbox)
+        # FIXME: https://github.com/astropy/astropy/issues/14705
+        # ValueError: operands could not be broadcast together with shapes (1100,) (1099,)
```

Comment:
Might point to a subtle bug deep within modeling bounding box machinery? 🤷

Reply:
Yeah, if someone knows how to properly fix it in Gaussian2D, that would be better.

Reply:
I think the fix has to be here somewhere:

https://github.com/astropy/astropy/blob/252b1c423838e2ae6d4585b178448d471513fc6c/astropy/modeling/utils.py#L179-L183

Reply:
I guess a more hacky alternative is to modify the test case so this won't happen for that test but it is not a real fix.

https://github.com/astropy/astropy/blob/252b1c423838e2ae6d4585b178448d471513fc6c/astropy/modeling/tests/example_models.py#L328

Reply:
I guess it really depends on the goal of the test.... If it was just a sanity check that basically the box was the same, with a slop of 1 pixel due to rounding allowed, then it would be fine to adjust it. But I don't know...

Reply:
The part of the test which is failing has nothing to do with the precise construction of the bounding box... Nor does it have anything to do with checking "if the box was the same"... Nor can it be the result of a "rounding error"..

The point of the test where it is failing is showing that a model properly discriminates if the inputs are inside/outside the bounding box when integrated into the model evaluation process. 

It specifically constricts two sets of inputs `x1, y1`, `x2, y2` such that:

1.  All the points in `x1, y1` are in `x2, y2` respectively
2. All points in `x2, y2` not in `x1, y1` lie outside the bounding box

The model is then evaluated on these points without/with bounding box enforcement respectively (to `x1, y1` vs `x2, y2`). This means that all the outputs corresponding to points outside the bounding box should be set to `np.NaN`. Consequently, due to the specific construction of `x2, y2` relative to `x1, y1` it should be the case that all the non `np.NaN` outputs of both evaluations should exactly correspond without any rounding errors (it is literally an exactly repeated computation).

To see that the assumptions should be satisfied, note that (1) is satisfied by
https://github.com/astropy/astropy/blob/e3b2444227a1bcb06931bc61cdd039936e421e8a/astropy/modeling/tests/test_models.py#L233-L249
because the construction of `x2, y2` by concatenating points onto `x1, y2` respectively.

While (2) is satisfied by the fact that the bounding box limits are used and the additional points are all created strictly `-/+idx*ddx` where `idx != 0`, where `-/+` is with respect to the lower/upper bounds of the bounding box. This implies that all the additional points are strictly outside the bounding box.  There should be no rounding error issues `ddx = .01`, which is much much larger than machine precision (so well outside any rounding error concerns). 

Since bounding boxes discriminate the input points lying outside the box using
https://github.com/astropy/astropy/blob/e3b2444227a1bcb06931bc61cdd039936e421e8a/astropy/modeling/bounding_box.py#L118-L131
It follows that the endpoints of the bounding box should lie within it (i.e. it defines a closed interval on each axis). So by construction all the inputs in `x1, y1` are inside the bounding box. Therefore, evaluating the model on `x1, y1` regardless of bounding box enforcement should result in no `np.NaN` outputs (`Gaussian2D` is well behaved on its bounding box intervals). While evaluating the model on `x2, y2` should result in `np.NaN` values for all the points not corresponding to those in `x1, y1`, while the corresponding points between `x1, y1` and `x2, y2` should all exactly match.

Note that, the error in question:
```
ValueError: operands could not be broadcast together with shapes (1100,) (1099,)
```
Indicates that some point in `x1, y1` is being evaluated to `np.NaN` in the `x2, y2` computation. There are a few possibilities:

1. The bounding box incorrectly discriminating on the endpoints of the bounding box as those points are specifically included in the computation. Meaning there is an issue with the `>` or `<` comparisons in the `outside` method.
2. Some sort of new numerical issue is causing a `np.NaN` occur when evaluating `Gaussian2D` on points contained within its bounding box (this is surprising since the default bounding box for `Gaussian2D` is selected specifically so that no numerical errors should occur when evaluating on points contained within it).
3. There is an issue with https://github.com/astropy/astropy/blob/e3b2444227a1bcb06931bc61cdd039936e421e8a/astropy/modeling/tests/test_models.py#L253 wherein  `~np.isnan(...)` is incorrectly flagging a non-`np.NaN` value as `NaN`.

The way I could distinguish between these possibilities would be

1.  Compare `inside_bbox` with  `inside_bbox[~np.isnan(inside_bbox)]` (eliminates/confirms possibility (2))
2. and/or Compute `inside_bbox` with the `with_bounding_box=True` option (eliminates/confirms possibility (1))
3. Manually examine the raw values of `inside_bbox` and `outside_bbox`  (eliminates/confirms possibility (3))

However, I can no-longer reproduce this issue, see  #14771 so I cannot exactly determine the cause.